app/utils/latex.ts

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/**
+ * Splits up latex code in an array consisting of each line of latex code.
+ * A latex linebreak is denoted by `\\`.
+ * A line is not broken in 2 cases:
+ * - The linebreak is in between brackets.
+ * - The linebreak occurs inbetween a `\begin` and `\end` statement.
+ */
+export function splitLatexCodeInLines(content: string): Array<string> {
+    let outLines = content.split('\\\\');
+
+    let i = 0;
+    while (i < outLines.length) {
+        if (testLatexLineBreak(outLines[i])) { //Line has no linebreak in between brackets
+            i += 1;
+        } else if (i < outLines.length - 2) {
+            outLines = outLines.slice(0, i).concat([outLines[i] + '\\\\' + outLines[i + 1]], outLines.slice(i + 2));
+        } else if (i === outLines.length - 2) {
+            outLines = outLines.slice(0, i).concat([outLines[i] + '\\\\' + outLines[i + 1]]);
+        } else {
+            return outLines;
+        }
+    }
+
+    return outLines.map((line) => line.trim());
+}
+
+function testLatexLineBreak(latexCode: string): boolean {
+    /**
+     * These checks are special because they need to check if the braces and statements are not escaped
+     */
+
+    //Check for cases
+    let beginCases = 0;
+    let endCases = 0;
+
+    let i = 0;
+    while (i < latexCode.length) {
+        const firstBegin = latexCode.indexOf('\\begin{', i);
```

Comment:
Unfortunaltely the react-native framework is not really that compatible with web library's. I looked into using a webView component to render the latex code but that was an ugly solution because inline this did not work well. So the `katex` library was not really an option on mobile

The `react-native-math-view library` is a way to fix this issue, but the drawback is that it does not support line breaks. So they need to be added manually

Reply:
I see what you mean. This feels like why I ended up writing so much of the mobile app's Markdown code from scratch. Unfortunately, react-native has a lot fewer libraries for things like this, so we end up having to do a lot ourselves.